hw/ip/aes/doc/dv/index.md

```diff
@@ -60,22 +64,60 @@ It can be created manually by invoking [`regtool`]({{< relref "util/reggen/READM
+* **generate_message_queue**: Generate a queue of randomized message items. 
+Each item will describe the parameters of a message but not hold any data, the data will be added later.
+This function does not call any sub-functions.
+* **send_msg_queue**: Take the queue of messages items and process them one by one.
+Send_msg_queue converts each message item into a queue of message transactions each called an aes_items by calling generate aes_item_queue().
+Then each message now described by a queue of items is processed by calling send_msg().
+* **generate_aes_item_queue**: Expands a message into a queue of a configuration item and N Data items where N = Message_length/block size.
+Data is randomized based on the constraints.
+* **send_msg**: Take a queue of configuration and data items and pass them to the AES for processing.
+This task will handle configuration of the core based on the configuration item.
+Then based on the test configuration it will do either a balanced or unbalanced processing of the data items.
+A balanced processing implies that for each input the task will wait until the resulting output have been read before attempting to write the next input.
+The unbalanced version will write the next data item as soon as the AES status register indicates that a new input will be accepted.
+In unbalanced mode the send_msg task also handles reading the output register.
+Knobs are available to control the balance between reads and writes.
+In either case the status_fsm task is called.
+* **status_fsm**: Read the status of the AES IP and based on the inputs return the status.
+When the task is in control of reading the output register it will poll the status until it indicates that the output is valid.
+It will also track the progress of processing detecting if something has gone differently than expected.
+If this happens it will try to recover.
+
+Using these higher level functions and tasks one can build a highly customized constraint random test without low level knowledge of the test environment and the DUT.
+
+Most tests use the aes_stress_vseq sequence as test sequence, and achieves different tests scenarios by using the constrained knobs to generate different behavior.
+
+
+#### Functional coverage'
 To ensure high quality constrained random stimulus, it is necessary to develop a functional coverage model.
 The following cover groups have been developed to prove that the test intent has been adequately met.
 
-WORK IN PROGRESS WILL COME WITH V2.
+* aes_ctrl_cg: this cover group checks that all parts of the control register has been exercised. 
+A few crosses is also included to ensure that we have seen all both forward and inverse for all modes.
+Similarly that we have tried all possible key_sizes in all modes, and that all have been tried both in auto and manual mode.
+More importantly for AES which is a security critical IP it also verifies that all illegal states has been seen.
```

Comment:
ha**ve** been seen ("illegal states" is plural). Similar for the next two lines ("have" and "were").

Reply:
thanks